pywinauto/__init__.py

```diff
@@ -104,23 +104,39 @@ def _get_com_threading_mode(module_sys):
 
 
     class Desktop(object):
-
         """Simple class to call something like ``Desktop().WindowName.ControlName.method()``"""
 
         def __init__(self, backend=None):
             """Create desktop element description"""
-            if backend:
-                self.backend = backend
-            else:
-                self.backend = backends.registry.name
+            if not backend:
+                backend = backends.registry.name
+            if backend not in backends.registry.backends:
+                raise ValueError('Backend "{0}" is not registered!'.format(backend))
+            self.backend = backends.registry.backends[backend]
 
-        def window(self, **criterion):
+        def window(self, **kwargs):
             """Create WindowSpecification object for top-level window"""
-            if 'top_level_only' not in criterion:
-                criterion['top_level_only'] = True
-            if 'backend' not in criterion:
-                criterion['backend'] = self.backend
-            return WindowSpecification(criterion)
+            if 'top_level_only' not in kwargs:
+                kwargs['top_level_only'] = True
+            if 'backend' not in kwargs:
+                kwargs['backend'] = self.backend.name
+            return WindowSpecification(kwargs)

```

Comment:
Hm... Are you sure it's necessary? Because in the Application.window() we have this:
```python
    win_spec = WindowSpecification(kwargs)
return win_spec
```

Reply:
Ok you are right). I check it out in IDLE, will fix this in Desktop.window() and Application.window().